sources/apollo/server.js

```diff
@@ -41,13 +44,23 @@ const server = new ApolloServer({
 	context: async ({ req }) => {
 		if (req.headers && req.headers.authorization) {
 		    const idToken=req.headers.authorization;
-		    try {
-				const decodedToken = await firebaseApp.auth().verifyIdToken(idToken)			
-				const {uid} = decodedToken;	
-				if(decodedToken.mongoID){
-					return {uid, permissions: await populatePermissions(decodedToken.mongoID)};
+			try {
+				console.time("firebase");
+
+				let userToken = tokenCache.get(idToken);
+				if (userToken === undefined) {
+					const decodedToken = await firebaseApp.auth().verifyIdToken(idToken)
+					// eslint-disable-next-line prefer-destructuring
+					userToken={ uid: decodedToken.uid, mongoID: decodedToken.mongoID }
+					tokenCache.set(idToken, userToken,36000);
+				}			
```

Comment:
Scenario
at t=0, Firebase minted JWT for the user
at t=30 min, App sent that token to the server
This caches the JWT for 60 mins, whereas it should cache only for 30 mins

Ideally
Store the exp of the JWT when you cache and check it also

Reply:
Great Suggestion. Will do it soon! 🚀